### ISSUE TITLE
Error Creating Athena Database

### ISSUE THREAD

**Issue by andrewmichael** [terraform-providers/terraform-provider-aws#4128]
### Terraform Version
Terraform 0.11.3
aws 1.14.0

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_athena_database

### Terraform Configuration Files
```hcl
resource "aws_athena_database" "pccp_datalake" {
  name          = "PccpDevDatalake"
  bucket        = "${aws_s3_bucket.pccp_datalake.bucket}"
  force_destroy = true
}
```

### Debug Output
https://gist.github.com/andrewmichael/4d1b1e5bf035b1e35264cd81111723c0

### Expected Behavior
Athena database is created, and subsequent plan / apply commands operate correctly.

### Actual Behavior
Plan output:-
```
Terraform will perform the following actions:

  + aws_athena_database.pccp_datalake
      id:            <computed>
      bucket:        "pccp-dev-datalake"
      force_destroy: "true"
      name:          "PccpDevDatalake"


Plan: 1 to add, 0 to change, 0 to destroy.
```
Apply end in error:-
```
Error: Error applying plan:

1 error(s) occurred:

* aws_athena_database.pccp_datalake: 1 error(s) occurred:

* aws_athena_database.pccp_datalake: [ERROR] Athena not found database: PccpDevDatalake, query result: default
pccpdevdatalake
```
Subsequent plan / apply fails.
```
Error: Error refreshing state: 1 error(s) occurred:

* aws_athena_database.pccp_datalake: 1 error(s) occurred:

* aws_athena_database.pccp_datalake: aws_athena_database.pccp_datalake: [ERROR] Athena not found database: PccpDevDatalake, query result: default
pccpdevdatalake
```
Observing the AWS console the database appears to have been created.
The only way out of this now is to manually deleting the Athena database, then revert to the previous version of remote state in S3.

Just reverting remote state, results in the following error:-
```
Error: Error applying plan:

1 error(s) occurred:

* aws_athena_database.pccp_datalake: 1 error(s) occurred:

* aws_athena_database.pccp_datalake: reason: FAILED: Execution Error, return code 1 from org.apache.hadoop.hive.ql.exec.DDLTask. Database PccpDevDatalake already exists

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`
1. `terraform apply`

### Important Factoids
IAM user has admin permissions
remote state is persisted in s3

**Comment by catsby** [comment]
I can't find it explicitly documented, but from looking and trying it out I'm seeing that the Athena service is lowercasing the `name`, so your mixed case example is failing the lookup. 

We need to add validation to the `name` attribute to check casing and enforce lower case (at least for now...), as anyone trying to use Athena will have a broken setup if they used mixed case.

**Comment by andrewmichael** [comment]
Wicked thanks, the one this is didn't try, will move to lower case.

:-)
Andy

**Comment by appilon** [comment]
Here are some docs on the allowable characters:
https://docs.aws.amazon.com/athena/latest/ug/tables-databases-columns-names.html

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with v1.15.0 of the AWS provider, likely mid next week. 👍
